cryptol-remote-api/python/cryptol/__init__.py

```diff
@@ -200,7 +200,7 @@ def process_result(self, res : Any) -> Any:
         if res['result'] == 'unsatisfiable':
             if self.qtype == 'sat':
                 return False
-            elif self.qtype == 'prove':
+            elif self.qtype == 'prove' or self.qtype == 'safe':
```

Comment:
This isn't the fault of your patch, but I'm somewhat leery about using strings here instead of an enumeration of some kind. Without looking, I wasn't sure whether `'prove'`, `'safe'`, and `'sat'` were the only possible options.

Reply:
This is a good point -- I think some type annotations using `Union[Literal['prove'], Literal['safe'], Literal['sat']]` will allow us to use strings (which is convenient for JSON compat) and be clear/precise.

Reply:
Does it even need to be that fancy? I was thinking of something like:

```py
class QueryType(str, Enum):
    PROVE = "prove"
    SAFE = "safe"
    SAT = "sat"
```

Reply:
It might be hard to convince me that my solution is more fancy than the `str` mixin `Enum` solution =) I'll give the enum a try.

Reply:
Admittedly, I'm not as familiar with `Literal`s as I am with `Enum`s. One of the reasons I would slightly prefer `Enum`s is that instead of writing `elif self.qtype == 'prove' or self.qtype == 'safe'`, you'd write `elif self.qtype == SmtQueryType.PROVE or self.qtype == SmtQueryType.SAT`. Although this is somewhat more verbose, it makes it clearer that the choices are coming from a fixed set of values rather than from arbitrary strings.

Ultimately, however, I'm not too opinionated on this. If you think the `Literal` approach would be better, I'll defer to your judgment.

Reply:
I agree with all of your intuitions/arguments/etc around enums here -- they do make things clearer in this case in many ways! =)